src/Tests.IntegrationTests/ClassifyMessageTests.cs

```diff
@@ -0,0 +1,86 @@
+using Microsoft.Extensions.Configuration;
+using System.Text.Json;
+using Tests.IntegrationTests.Models;
+
+namespace Tests.IntegrationTests
+{
+    public class ClassifyMessageTests
+    {
+        private readonly IConfiguration _configuration;
+
+        public ClassifyMessageTests()
+        {
+            _configuration = new ConfigurationBuilder()
+                .AddJsonFile(@"appsettings.json", false, false)
+                .Build();
+        }
+
+        [Fact(Timeout = 120000)]
+        public async Task GivenValidMessageReceivesValidResponse()
+        {
+            // Arrange
+            using var httpClient = new HttpClient();
+            var bot1Url = _configuration["Bot1Url"];
+            var chatsUri = new Uri($"{bot1Url}/client-api/chats");
+
+            // Act
+            // 1 Create Chat
+            var createdChat = await Request<Chat>(httpClient, Verb.Post, chatsUri).ConfigureAwait(false) as Chat;
+
+            // 2 Create Message
+            var chatMessageUri = new Uri($"{bot1Url}/client-api/chats/{createdChat.Id}/messages");
+            using var content = new StringContent("i want to register my child at school");
+            var createdChatMessage = await Request<ChatMessage>(httpClient, Verb.Post, chatMessageUri, content).ConfigureAwait(false) as ChatMessage;
+
+            // 3 Retry until we have 2 messages in the chat that this test created
+            Chat resultChat = null;
+            bool breakLoop = false;
+            while (!breakLoop)
+            {
+                var chats = await Request<List<Chat>>(httpClient, Verb.Get, chatsUri).ConfigureAwait(false) as List<Chat>;
+                resultChat = chats.First(c => c.Id == createdChat.Id);
+                if (resultChat.Messages.Count < 2)
+                {
+                    await Task.Delay(new TimeSpan(0, 0, 10)).ConfigureAwait(false);
```

Comment:
This is the nitty of picks - but TimeSpan.FromSeconds might be a bit more readable.